lib/flatpak-remote.c

```diff
@@ -814,9 +814,27 @@ flatpak_remote_commit (FlatpakRemote   *self,
   if (priv->local_collection_id_set)
     {
       if (priv->local_collection_id != NULL)
-        g_key_file_set_string (config, group, "collection-id", priv->local_collection_id);
+        {
+          g_key_file_set_string (config, group, "collection-id", priv->local_collection_id);
+
+          /* When a collection ID is set, flatpak uses signed per-repo and
+           * per-commit metadata instead of summary signatures. */
+          g_key_file_set_boolean (config, group, "gpg-verify-summary", FALSE);
+        }
       else
-        g_key_file_remove_key (config, group, "collection-id", NULL);
+        {
+          gboolean gpg_verify_value;
+
+          g_key_file_remove_key (config, group, "collection-id", NULL);
+
+          /* Without a collection ID gpg-verify-summary should go back to
+           * matching gpg-verify. */
+          gpg_verify_value = g_key_file_get_boolean (config, group, "gpg-verify", error);
+          if (*error == NULL)
```

Comment:
You can’t assume that `error` is non-`NULL`; this could be a `NULL` pointer dereference.

The idiomatic way of doing this kind of thing would be:
```
g_autoptr(GError) local_error = NULL;
gpg_verify_value = g_key_file_get_boolean (config, group, "gpg-verify", &local_error);
if (local_error == NULL)
  g_key_file_set_boolean (…);
```